src/cleanfigure.m

```diff
@@ -251,7 +246,7 @@ function pruneOutsideBox(meta, handle)
       id_remove  = id_remove(~idx);
   end
   % Replace the data points
-  replaceData(handle, id_replace, NaN(length(id_replace), size(data,2)));
```

Comment:
Is there a reason why you hoisted the `NaN` creation into the function? I think it makes more sense to have this generic `replaceData` function than to hardcode the function to replace with `NaN`. The new function is a function that you can implement by wrapping the old function with just 1 or 2 lines of new code.